File: silver/10845/c++/main.cpp
#include <iostream>
#include <queue>

int main()
{
    int num, data;
    std::queue<int> _queue;
    std::string command;
    std::cin >> num;

    for (int i = 0; i < num; i++) {
        std::cin >> command;
        if (command == "push") {
            std::cin >> data;
            _queue.push(data);
        } else if (command == "pop") {
            if (_queue.empty()) {
                std::cout << "-1\n";
            } else {
                std::cout << _queue.front() << '\n';
                _queue.pop();
            }
        } else if (command == "size") {
            std::cout << _queue.size() << '\n';
        } else if (command == "empty") {
            std::cout << _queue.empty() << '\n';
        } else if (command == "front") {
            if (_queue.empty()) {
                std::cout << "-1\n";
            } else {
                std::cout << _queue.front() << '\n';
            }
        } else if (command == "back") {
            if (_queue.empty()) {
                std::cout << "-1\n";
            } else {
                std::cout << _queue.back() << '\n';
            }
        }
    }

    return 0;
}

File: silver/10845/python/main.py
import sys
input = sys.stdin.readline

if (__name__ == "__main__"):

    num = int(input())
    queue = []
    for i in range(num):

        command = list(input().split())
        if (len(command) == 1):
            if (command[0] == "pop" and len(queue) > 0):
                print(queue.pop(0))
            elif (command[0] == "size"):
                print(len(queue))
            elif (command[0] == "empty"):
                if (len(queue) == 0): print(1)
                else : print(0)
            elif (command[0] == "front" and len(queue) > 0):
                print(queue[0])
            elif (command[0] == "back" and len(queue) > 0):
                print(queue[-1])
            else :
                print(-1)
        else :
            queue.append(int(command[1]))